stdlib/char.ml

```diff
@@ -63,3 +63,60 @@ external seeded_hash_param :
   int -> int -> int -> 'a -> int = "caml_hash" [@@noalloc]
 let seeded_hash seed x = seeded_hash_param 10 100 seed x
 let hash x = seeded_hash_param 10 100 0 x
+
+module Ascii = struct
+
+  (* Characters *)
+
+  let min = '\x00'
+  let max = '\x7F'
+
+  (* Predicates *)
+
+  let is_valid = function '\x00' .. '\x7F' -> true | _ -> false
+  let is_upper = function 'A' .. 'Z' -> true | _ -> false
+  let is_lower = function 'a' .. 'z' -> true | _ -> false
+  let is_letter = function 'A' .. 'Z' | 'a' .. 'z' -> true | _ -> false
+  let is_alphanum = function
+    | '0' .. '9' | 'A' .. 'Z' | 'a' .. 'z' -> true | _ -> false
+
+  let is_white = function ' ' | '\t' .. '\r'  -> true | _ -> false
+  let is_blank = function ' ' | '\t' -> true | _ -> false
+  let is_graphic = function '!' .. '~' -> true | _ -> false
+  let is_print = function ' ' .. '~' -> true | _ -> false
+  let is_control = function '\x00' .. '\x1F' | '\x7F' -> true | _ -> false
+
+  (* Decimal digits *)
+
+  let is_digit = function '0' .. '9' -> true | _ -> false
+  let digit_to_int = function
+    | '0' .. '9' as c -> code c - 0x30
+    | c -> invalid_arg (escaped c ^ ": not a decimal digit")
+
+  let digit_of_int n = unsafe_chr (0x30 + (n mod 10))
```

Comment:
Shouldn't we handle the case of `n < 0` here? Eg by doing
```
let n = n mod 10 in
let n = if n < 0 then n + 10 else n in
...
```

Reply:
Should we do something similar in `lower_hex_digit_of_int` and `upper_hex_digit_of_int`?

Reply:
It seems one of my comments didn't get trough: @nojb I don't think your code does what it does, I simply used `abs`. 

@yallop yes, I'll do that in the name of least surprise. The other functions worked differently because in my [blueprint](https://erratique.ch/software/b0/doc/B0_std/Char/Ascii/index.html) I didn't have the decimal ones.

Reply:
The latest commit does that. I also tweaked the doc strings to mention the `abs` to make it clear the functions work on negative numbers aswell.